handlers.go

```diff
@@ -104,15 +104,37 @@ func (m *MockOIDC) Token(rw http.ResponseWriter, req *http.Request) {
 		return
 	}
 
-	if !m.validateTokenParams(rw, req) {
-		return
-	}
-
-	code := req.Form.Get("code")
-	session, err := m.SessionStore.GetSessionByID(code)
-	if err != nil {
-		errorResponse(rw, invalidGrant, fmt.Sprintf("Invalid code: %s", code),
-			http.StatusUnauthorized)
+	var session *Session
+	grantType := req.Form.Get("grant_type")
+	switch grantType {
+	case "authorization_code":
+		if !m.validateTokenParams(rw, req) {
+			return
+		}
+		code := req.Form.Get("code")
+		session, err = m.SessionStore.GetSessionByID(code)
+		if err != nil {
+			errorResponse(rw, invalidGrant, fmt.Sprintf("Invalid code: %s", code),
+				http.StatusUnauthorized)
+			return
+		}
+	case "refresh_token":
+		if !m.validateRefreshParams(rw, req) {
+			return
+		}
+		refreshToken := req.Form.Get("refresh_token")
+		token, ok := m.validateTokenString(refreshToken, rw, req)
+		if !ok {
+			return
+		}
+		session, err = m.SessionStore.GetSessionByToken(token)
+		if err != nil {
+			errorResponse(rw, invalidGrant, "Invalid refresh token",
```

Comment:
I think `invalidRequest` as the error type if the refresh token can't be validated? I'm not positive, I need to see what codes are used when auth fails.